src/app/loadout/plug-drawer/PlugDrawer.tsx

```diff
@@ -190,17 +233,50 @@ export default function PlugDrawer({
       sheetClassName="item-picker"
       freezeInitialHeight={true}
     >
-      {groupedPlugs.map((plugItem) => (
+      {queryFilteredPlugSets.map((plugsWithMaxSelectable) => (
         <PlugSection
-          key={plugItem[0].plug.plugCategoryHash}
-          plugs={plugItem}
-          selected={selected}
+          key={plugsWithMaxSelectable.plugSetHash}
+          plugsWithMaxSelectable={plugsWithMaxSelectable}
+          selected={selected[plugsWithMaxSelectable.plugSetHash] || []}
           displayedStatHashes={displayedStatHashes}
-          isPlugSelectable={(plug) => isPlugSelectable(plug, selected)}
+          isPlugSelectable={(plug) =>
+            isPlugSelectable(plug, _.compact(Object.values(selected).flat()))
+          }
           onPlugSelected={onPlugSelected}
           onPlugRemoved={onPlugRemoved}
         />
       ))}
     </Sheet>
   );
 }
+
+/**
+ * This creates the internally used state for the selected plugs.
+ * We want to split the selected plugs up into groups based on the plugSetHash we attribute them too.
```

Comment:
Why?

Reply:
It's so we know if a plug belongs to the artificer group or the class item socket group. I couldn't think of a better way to do it. Will document.